packages/vite/src/node/server/ws.ts

```diff
@@ -30,10 +32,30 @@ export function createWebSocketServer(
       }
     })
   } else {
+    let websocketServerOptions: WebSocket.ServerOptions = {}
+    const port = (hmr && hmr.port) || 24678
```

Comment:
**suggestion (non-blocking):** This can be shortened
```suggestion
    const port = hmr?.port || 24678
```

Reply:
Tried this but it doesn't typecheck. The value of `hmr` can be an options object or `false`, and the `?` operator will allow the property access of both of those, because both of those aren't undefined. `false.port` doesn't typecheck. What we'd need is a `|.` operator that only tried to do property access if the value was truthy. `&&` seems to suffice here.

Reply:
Haha :smile: interesting proposal `|.`
Yeah sorry for such inconvenient reviews, but I review in the GitHub Web UI and there I don't have type-checking
I hope we can improve this with #2613, then eslint will show up warnings in the Web UI via #2622